huaweicloud/data_source_huaweicloud_vpc_subnet_test.go

```diff
@@ -21,59 +26,83 @@ func TestAccVpcSubnetV1DataSource_basic(t *testing.T) {
+					resource.TestCheckResourceAttr(dNameByCIDR, "gateway_ip", gateway),
+					resource.TestCheckResourceAttr(dNameByCIDR, "status", "ACTIVE"),
+					resource.TestCheckResourceAttr(dNameByCIDR, "dhcp_enable", "true"),
+					testAccCheckVpcSubnetV1DataSourceID(dNameByName),
+					resource.TestCheckResourceAttr(dNameByName, "name", rName),
+					resource.TestCheckResourceAttr(dNameByName, "cidr", cidr),
+					resource.TestCheckResourceAttr(dNameByName, "gateway_ip", gateway),
+					resource.TestCheckResourceAttr(dNameByName, "status", "ACTIVE"),
+					resource.TestCheckResourceAttr(dNameByName, "dhcp_enable", "true"),
+					testAccCheckVpcSubnetV1DataSourceID(dNameByVpcID),
+					resource.TestCheckResourceAttr(dNameByVpcID, "name", rName),
+					resource.TestCheckResourceAttr(dNameByVpcID, "cidr", cidr),
+					resource.TestCheckResourceAttr(dNameByVpcID, "gateway_ip", gateway),
+					resource.TestCheckResourceAttr(dNameByVpcID, "status", "ACTIVE"),
+					resource.TestCheckResourceAttr(dNameByVpcID, "dhcp_enable", "true"),
 				),
 			},
 		},
 	})
 }
 
-func testAccDataSourceVpcSubnetV1Check(n, name, cidr, gateway_ip string) resource.TestCheckFunc {
+func TestAccVpcSubnetV1DataSource_ipv6(t *testing.T) {
+	rName := fmt.Sprintf("tf-acc-test-%s", acctest.RandString(5))
+	tmp := strconv.Itoa(acctest.RandIntRange(1, 254))
+	cidr := fmt.Sprintf("172.16.%s.0/24", string(tmp))
+	gateway := fmt.Sprintf("172.16.%s.1", string(tmp))
+	dName := "data.huaweicloud_vpc_subnet.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccVpcSubnetV1DataSource_ipv6(rName, cidr, gateway),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVpcSubnetV1DataSourceID(dName),
+					resource.TestCheckResourceAttr(dName, "name", rName),
+					resource.TestCheckResourceAttr(dName, "cidr", cidr),
+					resource.TestCheckResourceAttr(dName, "gateway_ip", gateway),
+					resource.TestCheckResourceAttr(dName, "status", "ACTIVE"),
+					resource.TestCheckResourceAttr(dName, "dhcp_enable", "true"),
+					resource.TestCheckResourceAttr(dName, "ipv6_enable", "true"),
+					resource.TestMatchResourceAttr(dName, "ipv6_cidr", regexp.MustCompile("([[:xdigit:]]*:){5}[[:xdigit:]]*/\\d{1,3}")),
+					resource.TestMatchResourceAttr(dName, "ipv6_gateway", regexp.MustCompile("([[:xdigit:]]*:){5}([[:xdigit:]]){1,3}")),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckVpcSubnetV1DataSourceID(n string) resource.TestCheckFunc {
 	return func(s *terraform.State) error {
 		rs, ok := s.RootModule().Resources[n]
 		if !ok {
-			return fmt.Errorf("root module has no resource called %s", n)
-		}
-
-		subnetRs, ok := s.RootModule().Resources["huaweicloud_vpc_subnet.test"]
-		if !ok {
-			return fmt.Errorf("can't find huaweicloud_vpc_subnet.test in state")
+			return fmt.Errorf("Can't find vpc subnet connection data source: %s", n)
```

Comment:
how to understand `connection` here?

Reply:
This description is confusing to read. Updation has completed.